alpaca/data/historical/option.py

```diff
@@ -0,0 +1,333 @@
+from collections import defaultdict
+from enum import Enum
+from typing import Dict, List, Optional, Union
+
+from alpaca.common.constants import DATA_V2_MAX_LIMIT
+from alpaca.common.enums import BaseURL
+from alpaca.common.rest import RESTClient
+from alpaca.common.types import RawData
+from alpaca.data.historical.utils import (
+    format_dataset_response,
+    format_latest_data_response,
+    format_snapshot_data,
+    parse_obj_as_symbol_dict,
+)
+from alpaca.data.models.quotes import Quote, QuoteSet
+from alpaca.data.models.snapshots import Snapshot
+from alpaca.data.models.trades import Trade, TradeSet
+from alpaca.data.requests import (
+    OptionChainRequest,
+    OptionLatestQuoteRequest,
+    OptionLatestTradeRequest,
+    OptionQuotesRequest,
+    OptionSnapshotRequest,
+    OptionTradesRequest,
+)
+
+
+class DataExtensionType(Enum):
+    """Used to classify the type of endpoint path extensions"""
+
+    LATEST = "latest"
+    SNAPSHOT = "snapshot"
+
+
+class OptionHistoricalDataClient(RESTClient):
+    """
+    The REST client for interacting with Alpaca Market Data API option data endpoints.
+
+    Learn more on https://docs.alpaca.markets/docs/about-market-data-api
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        secret_key: Optional[str] = None,
+        oauth_token: Optional[str] = None,
+        use_basic_auth: bool = False,
+        raw_data: bool = False,
+        url_override: Optional[str] = None,
+    ) -> None:
+        """
+        Instantiates a Historical Data Client.
+
+        Args:
+            api_key (Optional[str], optional): Alpaca API key. Defaults to None.
+            secret_key (Optional[str], optional): Alpaca API secret key. Defaults to None.
+            oauth_token (Optional[str]): The oauth token if authenticating via OAuth. Defaults to None.
+            use_basic_auth (bool, optional): If true, API requests will use basic authorization headers.
+            raw_data (bool, optional): If true, API responses will not be wrapped and raw responses will be returned from
+              methods. Defaults to False. This has not been implemented yet.
+            url_override (Optional[str], optional): If specified allows you to override the base url the client points
+              to for proxy/testing.
+        """
+        super().__init__(
+            api_key=api_key,
+            secret_key=secret_key,
+            oauth_token=oauth_token,
+            use_basic_auth=use_basic_auth,
+            api_version="v1beta1",
+            base_url=url_override if url_override is not None else BaseURL.DATA,
+            sandbox=False,
+            raw_data=raw_data,
+        )
+
+    def get_option_quotes(
```

Comment:
This is not implemented yet on the server side.

Reply:
Thank you! commented out until implemented on the server side.